tests/Unit/NumericalAlgorithms/LinearOperators/Test_IndefiniteIntegral.cpp

```diff
@@ -0,0 +1,113 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/StripeIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/LinearOperators/IndefiniteIntegral.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t Dim, typename T>
+DataVector integrand(const std::array<size_t, Dim>& exponent,
+                     const T& x) noexcept {
+  DataVector integrand(get<0>(x).size(), 1.0);
+  for (size_t d = 0; d < Dim; ++d) {
+    integrand *= pow(x.get(d), gsl::at(exponent, d));
+  }
+  return integrand;
+}
+
+template <size_t Dim, typename T>
+DataVector integral(const std::array<size_t, Dim>& exponent, const T& x,
+                    const Mesh<Dim> mesh,
+                    const size_t only_this_dim = 100) noexcept {
+  DataVector integral(get<0>(x).size(), 1.0);
+  for (size_t d = 0; d < Dim; ++d) {
+    if (only_this_dim == 100 or d == only_this_dim) {
+      integral *= pow(x.get(d), gsl::at(exponent, d) + 1) /
+                  (gsl::at(exponent, d) + 1.0);
+      // Adjust to zero integral on lower boundary.
+      for (StripeIterator stripe_it(mesh.extents(), d); stripe_it;
+           ++stripe_it) {
+        for (size_t count = 1, i = stripe_it.offset() + stripe_it.stride();
+             count < mesh.extents()[d]; ++count, i += stripe_it.stride()) {
+          integral[i] -= integral[stripe_it.offset()];
+        }
+        integral[stripe_it.offset()] = 0.0;
+      }
+    } else {
+      integral *= pow(x.get(d), gsl::at(exponent, d));
```

Comment:
Once we work out what your goals are for the path-like integral, I might ask for a non-separable test.